invenio_deposit/api.py

```diff
@@ -308,3 +308,13 @@ def sort_by(*args, **kwargs):
             files_.sort_by = sort_by
 
         return files_
+
+    def get_file_by_id(self, file_id):
```

Comment:
It's solved on routing level. It will try to find the file by `file_id` and returns its `key` in `request.view_args`.